README.md

```diff
@@ -246,18 +467,46 @@ These are the options available in the `bsconfig.json` file.
             "codes": [1000, 1011] //ignore these specific codes from vendor libraries
         }]
         ```
- - **diagnosticLevel**: `'hint' | 'info' | 'warn' | 'error'` - Specify what diagnostic levels are printed to the console. This has no effect on what diagnostics are reported in the LanguageServer. Defaults to 'warn'
 
- - **autoImportComponentScript**: `bool` - BrighterScript only: will automatically import a script at transpile-time for a component with the same name if it exists.
+Defaults to `undefined`.
+
+#### `diagnosticLevel`
+
+Type: `"hint" | "info" | "warn" | "error"`
+
+Specify what diagnostic levels are printed to the console. This has no effect on what diagnostics are reported in the LanguageServer. Defaults to `"warn"`.
+
+#### `autoImportComponentScript`
+
+Type: `bool`
+
+BrighterScript only: will automatically import a script at transpile-time for a component with the same name if it exists. Defaults to `false`.
+
+#### `sourceRoot`
+
+Type: `string`
+
+Override the root directory path where debugger should locate the source files. The location will be embedded in the source map to help debuggers locate the original source files. This only applies to files found within `rootDir`. This is useful when you want to preprocess files before passing them to BrighterScript, and want a debugger to open the original files. This option also affects the `SOURCE_FILE_PATH` and `SOURCE_LOCATION` source literals.
+
+#### `plugins`
+
+Type: `Array<string>`
+
+List of node scripts or npm modules to load as plugins to the BrighterScript compiler. Defaults to `undefined`.
+
+If a child bsconfig extends from a parent bsconfig, and both bsconfigs specify a `plugins` field, the child's `plugins` field will completely overwrite the parent's `plugins` field.
```

Comment:
I added this line because this confused me at first. I tested this but it wouldn't be a bad idea to confirm that I'm not missing something.

Reply:
This is technically correct, but it's not limited to just `plugins`. We're doing a "replace every property in parent with properties found in child". so it would also apply to `files`, `diagnosticFilters`, etc... Might be worth explaining it a bit more generically, or making this a specific example. Perhaps something like this:

>Note: child config properties completely replace parent config properties. For example: ifa child bsconfig extends from a parent bsconfig, and both bsconfigs specify a `plugins` field, the child's `plugins` field will completely overwrite the parent's `plugins` field.

Reply:
I wanted to provide this information in general while also making it easy for people who are only looking at the documentation for the specific field they're stuck on, so in the new version I added the general explanation to the `extends` field and specifically called it out in the `files`, `plugins`, `diagnosticFilters`, and `require` fields. Its a lot of duplication but I'm hoping it will be more helpful to people who are quickly skimming.